radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/handler/ArchiveNetworkHandler.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * (C) Copyright 2021 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.api.handler;
+
+import org.json.JSONObject;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.radixdlt.api.service.NetworkInfoService;
+import com.radixdlt.universe.Magic;
+
+import static com.radixdlt.api.JsonRpcUtil.jsonObject;
+import static com.radixdlt.api.JsonRpcUtil.response;
+
+@Singleton
```

Comment:
We decided early on not to use @Singleton so that this can be managed only by Modules themselves.

Reply:
Sorry, I was not involved in that decision. Anyway, fixed.